blueprint/core/blueprint_EcmascriptEngine.cpp

```diff
@@ -556,12 +446,23 @@ namespace blueprint
                                 pushVarToDukStack(ctx, args.arguments[i]);
 
                             // Invocation
-                            if (duk_pcall(ctx, args.numArguments) != DUK_EXEC_SUCCESS)
-                                duk_throw(ctx);
+                            try {
+                                detail::safeCall(ctx, args.numArguments);
+                            } catch (Error const& err) {
+                                internalErrorHandler();
+                                throw err;
+                            }
 
                             // Clean the result and the stash off the top of the stack
                             duk_pop_2(ctx);
 
+                            // Here too, we know that we have potentially just pushed temporaries
+                            // to the stack. Duktape will eventually get around to garbage collection,
+                            // but specifically where we have LambdaHelper allocations for these
+                            // temporaries, we want to keep our release pool fairly small. For that
+                            // reason, we manually invoke a garbage collection pass right away.
+                            duk_gc(ctx, 0);
```

Comment:
Cool. I wonder if it would be worth adding some little helper function wrapped in an `#ifdef DEBUG_RELEASE_POOL` or something which dumps the size of the release pool at certain points and prints whenever `removeLambdaHelper` is called ? Might be a useful thing to toggle on whenever similar such performance issues arise.

Reply:
Yea I had written that actually, but I removed it because I was seeing a constant release pool size in the GainPlugin example with the canvas animation, so the repetitive logs became not much value after the first few lines.

Think it'd be worth adding back in still?

Reply:
It might not be that essential now. Assuming the weird issues I'm seeing currently aren't to do with the release pool!

Reply:
Hmm. Maybe not. I think the duk_context dump wrapped under some ifdef toggle could certainly be useful for user bug reports. Maybe some release pool dump out is still useful? I think a separate PR for more debug tools maybe rather than bloating out this set of changes though

Reply:
Ok yea, sounds good! I think your app is more complicated than what I'm working against, so I'd be curious to hear from you as to what info you'd want to see in that dump out.

Reply:
Sure. I'll have a play. Dumping the context and checking it's state certainly helped me resolve a few errors last time I was digging through here.